microsoft/testsuites/hpc/infinibandsuit.py

```diff
@@ -100,3 +101,49 @@ def verify_hpc_over_nd(self, log: Logger, node: Node) -> None:
             assert_that(modprobe.is_module_loaded(module)).described_as(
                 f"Module {module} is not loaded."
             ).is_true()
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Identify the infiniband devices and their cooresponding network interfaces
+        2. Run several ping-pong tests to check RDMA / Infiniband functionality
+        """,
+        priority=4,
+        requirement=simple_requirement(
+            supported_features=[Infiniband],
+            min_count=2,
+        ),
+    )
+    def verify_ping_pong(self, environment: Environment, log: Logger) -> None:
+        # Constants
+        ping_pong_tests = ["ibv_rc_pingpong", "ibv_uc_pingpong", "ibv_ud_pingpong"]
+
+        server_node = environment.nodes[0]
+        client_node = environment.nodes[1]
+
+        infiniband = server_node.features[Infiniband]
+        ib_interfaces = infiniband.get_ib_interfaces()
+
+        for interface in ib_interfaces:
+            ib_device_name = interface[0]
+            ip_addr = interface[2]
+
+            for test in ping_pong_tests:
+                server_process = server_node.execute_async(
+                    f"{test} -d {ib_device_name}"
+                )
+                client_process = client_node.execute_async(
+                    f"{test} -d {ib_device_name} {ip_addr}"
+                )
+                server_result = server_process.wait_result()
```

Comment:
need turn off firewall before this testing.

Reply:
Please check if it needs to turn off fire wall. There is a tool called Firewall, it's easy to use by call `stop`.

Reply:
This is the process recommended for HPC machines
``` bash
systemctl stop iptables.service
systemctl disable iptables.service
systemctl mask firewalld
systemctl stop firewalld.service
systemctl disable firewalld.service
iptables -nL
sed -i -e's/SELINUX=enforcing/SELINUX=disabled/g' /etc/selinux/config
```

Reply:
The test target of this test case is about the IB, so whatever the way to turn off the firewall is ok here. If the current Firewall tool cannot turn off correctly, please fix it.